pkgs/tools/security/dnspeep/default.nix

```diff
@@ -0,0 +1,31 @@
+{ fetchFromGitHub, lib, libpcap, rustPlatform, stdenv }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "dnspeep";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "jvns";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0lz22vlgi1alsq676q4nlzfzwnsrvziyqdnmdbn00rwqsvlb81q6";
+  };
+
+  cargoLock = {
+    # update Cargo.lock every update
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "pcap-0.8.1" = "1nnfyhlhcpbl4i6zmfa9rvnvr6ibg5khran1f5yhmr9yfhmhgakd";
+    };
+  };
```

Comment:
Why can't we use the upstream one?

Reply:
using `fetchurl` as `cargoLock.lockFile` utilizes ifd and upsets ofborg

Reply:
The upstream repo has a lock file included.

Reply:
`cargoLock` is an alternative to `cargoSha256`, not `Cargo.lock`. upsream lock file includes a git dependency, which doesn't work with `cargoSha256`, so I have to download `Cargo.lock` and use `cargoLock` as an alternative

Reply:
Can you add a comment that explains that the git dependency requires this? Thank you!

Reply:
added